### ISSUE TITLE
lua-language-server can't write/read settings file

### ISSUE THREAD

**Issue by GitSparTV** [sublimelsp/LSP#1498]
**Describe the bug**
I'm using `sumneko/lua-language-server` and I encountered a problem when Code Actions doesn't change anything.
Anything settings-file related. In VSCode (since this lsp was made for vscode but allows integrations) the settings are saved into `filedir/.vscode/settings.json`.

**To Reproduce**
Activate this LSP and make undefined global or language specific things.
Example:
```lua
local a = 1LL
```

**Expected behavior**
Settings should be saved and affect lsp output.

**Screenshots**

**Environment:**
- OS: Windows x64
- Sublime Text version: 3.2.2 build 3211
- LSP version: `0.14.3`
- Language servers used: sumneko/lua-language-server

**Comment by rwols** [comment]
Yep. You'd need to catch the workspace/executeCommand request and handle it ST-side (client-side). There's no code for this in 0.14.3, but more recent versions do allow this. I have something here: https://github.com/sublimelsp/LSP-lua/blob/main/plugin.py it'll only work for ST4 though. And it doesn't yet handle modifying settings like you describe. But it should catch the command in a callback called `on_pre_server_command`. Here is a good example of that: https://github.com/sublimelsp/LSP-jdtls/blob/1e0ad49960bd72b3f321ce489909da2bc50f2d05/plugin.py#L47-L61 (also only works on ST4).

**Comment by GitSparTV** [comment]
So there is no way to make it work on ST3?

**Comment by rwols** [comment]
It's possible from a technical point of view to port all of the changes required for this `on_pre_server_command` callback that were added in the ST4 branch to the ST3 branch. But I don't see myself doing that. I will transfer this issue to the LSP-lua repository as it seems fitting there.
